### PR TITLE
Fix split op in the way it deals with the optional input

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: As title

**Motivation and Context**
Resolve https://github.com/microsoft/onnxruntime/issues/7075

### CODE REVIEW

**Review by xzhu1900 — DISMISSED** [review]

File: onnxruntime/test/providers/cpu/tensor/split_op_test.cc [hunk 1]
```diff
@@ -613,5 +616,25 @@ TEST(SplitOperatorTest, Uint8NegativeAxis) {
   RunTest<uint8_t>(axis, {}, input, outputs, false);
 }
 
+TEST(SplitOperatorTest, MissingOptionalInputAdded) {
```
Comment (xzhu1900) [inline onnxruntime/test/providers/cpu/tensor/split_op_test.cc:619]: This probably is out the scope of this change. But just curious, based on the spec, if the optional `split` input is mssing, the tensor is split to equal sized parts, but what if the dimension in that axis cannot be equally divided into 2? Will ORT error out in this case?

Comment (hariharans29) [reply onnxruntime/test/providers/cpu/tensor/split_op_test.cc:619]: Yes, this is enforced on line 74 in PrepareForCompute() (and yes a failure status will be returned back to the user)

  if (split_sizes.empty()) {
    // equal split based on number of outputs
    if (split_dim_size % static_cast<size_t>(num_outputs) != 0) {
      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Input cannot be split evenly on selected axis. Input shape=", input_shape,
                             " Axis=", axis_, " NumOutputs=", num_outputs);
    }

**Comment by hariharans29** [comment]
@xzhu1900 - could you please re-sign off again if all looks okay ? Had to make a minor correction in the cuda ep file.

**Review by xzhu1900 — APPROVED** [review]

**Comment by xzhu1900** [comment]
signed off. Thanks for fixing this :-)

---
In reply to: [817953145](https://github.com/microsoft/onnxruntime/pull/7302#issuecomment-817953145) [](ancestors = 817953145)

**Pull request merged**